### PR TITLE
Add a tutorial section

### LINKED ISSUE #382: [JOSS] Adding tutorials and link to documentation
Hi alchemlyb team,

This is just to organize the comments from https://github.com/openjournals/joss-reviews/issues/6934 related to the repo: 

* Add an example case with input files to walk through the main functionalities 
* Add a link to docs inside README

Thanks and best wishes.

### PR DESCRIPTION
**Pull request by xiki-tempula**
Fix #382 

- add link to docs in README
- add tutorial section to docs

See https://github.com/openjournals/joss-reviews/issues/6934 :

- providing a single example case with input file, e.g. simple example, e.g., a self-contained jupyter notebook, to walk through the main functionalities.
- add link to docs inside README

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+This will generate a plot that looks like the one below:
+
+.. figure:: images/O_MBAR.png
+
+Automated workflow
+------------------
+
+The above workflow can be automated using the :mod:`~alchemlyb.workflows.ABFE`
+which provides a good practice for using **alchemlyb** in a end-to-end fashion.
```
Comment (orbeckst) [inline docs/tutorial.rst:113]:
```suggestion
The above manual workflow can be automated using the :mod:`~alchemlyb.workflows.ABFE`
workflow class, which demonstrates a best practice implementation for using **alchemlyb** in a end-to-end fashion.

.. SeeAlso::
   See the :ref:`ABFE workflow <abfe-workflow>` section of the documentation for
   more details on how to use :mod:`~alchemlyb.workflows.ABFE`.
   
*Absolute binding free energy* (ABFE) calculations can be analyzed with
two lines of code in a fully automated manner.
All parameters are set when invoking :class:`~alchemlyb.workflows.abfe.ABFE`
and reasonable defaults are chosen for any parameters not set explicitly. The two steps are to

1. initialize an instance of the :class:`~alchemlyb.workflows.abfe.ABFE` class
2. invoke the :meth:`~alchemlyb.workflows.ABFE.run` method to execute the
   complete workflow.
   
As an example, we again use data from a GROMACS ABFE simulation that is available
in alchemtest_. In this case, executing the workflow would look similar
to the following code ::

    >>> from alchemtest.gmx import load_ABFE
    >>> from alchemlyb.workflows import ABFE
    >>> # Obtain the path of the data
    >>> import os
    >>> dir = os.path.dirname(load_ABFE()['data']['complex'][0])
    >>> print(dir)
    'alchemtest/gmx/ABFE/complex'
    >>> workflow = ABFE(units='kcal/mol', software='GROMACS', dir=dir,
    >>>                 prefix='dhdl', suffix='xvg', T=298, outdirectory='./')
    >>> workflow.run(skiptime=10, uncorr='dhdl', threshold=50,
    >>>              estimators=('MBAR', 'BAR', 'TI'), overlap='O_MBAR.pdf',
    >>>              breakdown=True, forwrev=10)


The workflow uses :mod:`~alchemlyb.parsing` to parse the data from the
energy files, remove the initial un-equilibrated frames and decorrelate the data
with :mod:`~alchemlyb.preprocessing.subsampling`. The decorrelated datasets
:ref:`dHdl <dHdl>` and :ref:`u_nk <u_nk>` are then passed to
:mod:`~alchemlyb.estimators` for free energy estimation. The workflow will
also perform a set of analysis that allows the user to examine the quality of
the estimation.
   
```

**Review by orbeckst — APPROVED** [review]
Looking good, I am just making small edits and add a bit more content to workflows (copied and shortened from docs).

**Comment by orbeckst** [comment]
@ryankzhu would you like to review the added tutorial?

The built docs are at https://alchemlyb--379.org.readthedocs.build/en/379/tutorial.html

**Comment by orbeckst** [comment]
@xiki-tempula let's wait with merging for at least 2 days to give our JOSS reviewers a chance to look at it and ask for changes.

**Comment by ryankzhu** [comment]
Thank you for adding the tutorial section to the docs! The tutorial looks good to me.

**Pull request merged**